### PR TITLE
PUB-2291-Updated OPA lists

### PR DESCRIPTION
**Pull request by MindGames2015**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

https://tools.hmcts.net/jira/browse/PUB-2291

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPublicListHelper.java [hunk 1]
```diff
@@ -75,20 +78,33 @@ private static CaseInfo buildHearingCase(JsonNode hearingCase) {
         return new CaseInfo(
             GeneralHelper.findAndReturnNodeText(hearingCase, CASE_URN),
             scheduledHearingDate,
-            GeneralHelper.formatNodeArray(hearingCase, REPORTING_RESTRICTION_DETAIL, DELIMITER)
+            formatReportingRestriction(hearingCase)
         );
     }
 
-    private static List<Defendant> processPartyRoles(JsonNode hearing) {
+    private static String formatReportingRestriction(JsonNode node) {
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPublicListHelper.java:85]: What is the difference between this method and the original one used? The original one is GeneralHelper.formatNodeArray(o, REPORTING_RESTRICTION_DETAIL, DELIMITER) and it looks identical to the new one you created. If possible we should reuse the same common method

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPressListHelper.java [hunk 2]
```diff
@@ -106,18 +104,31 @@ private static OpaCaseInfo buildHearingCase(JsonNode hearingCase) {
         return new OpaCaseInfo(
             GeneralHelper.findAndReturnNodeText(hearingCase, CASE_URN),
             scheduledHearingDate,
-            GeneralHelper.formatNodeArray(hearingCase, REPORTING_RESTRICTION_DETAIL, DELIMITER)
+            formatReportingRestriction(hearingCase)
         );
     }
 
-    private static List<OpaDefendantInfo> processPartyRoles(JsonNode hearing) {
+    private static String formatReportingRestriction(JsonNode node) {
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPressListHelper.java:111]: Same as the public list comment. You can just use the original common method

File: src/test/resources/mocks/opaPressList.json [hunk 3]
```diff
@@ -34,6 +34,125 @@
...
+                                      "offenceSection": "Offence section 1",
+                                      "offenceWording": "Offence wording 1",
+                                      "plea": "NOT_GUILTY",
+                                      "pleaDate": "2023-09-20T09:00:00.123Z"
+                                    }
+                                  ]
+                                }
+                              },
+                              {
+                                "partyRole": "DEFENDANT",
```
Comment (KianKwa) [inline src/test/resources/mocks/opaPressList.json:69]: This defendant and the 2 below is pretty much identical. There is no need to repeatedly testing the same thing. You can remove 2 of them.

File: src/test/resources/mocks/opaPressList.json [hunk 4]
```diff
@@ -45,131 +164,131 @@
...
-                                  "offenceTitle": "Offence title 2",
-                                  "offenceSection": "Offence section 2",
-                                  "offenceWording": "Offence wording 2",
-                                  "plea": "NOT_GUILTY",
-                                  "pleaDate": "2023-09-22T09:00:00.123Z",
-                                  "reportingRestrictionDetail": [
-                                    "Offence reporting restriction detail 1"
+                              },
+                              {
+                                "partyRole": "DEFENDANT",
```
Comment (KianKwa) [inline src/test/resources/mocks/opaPressList.json:231]: Same here 3 very similar cases. Can remove 2 of them

File: src/test/resources/mocks/opaPublicList.json [hunk 5]
```diff
@@ -44,114 +140,210 @@
                           },
                           {
                             "caseUrn": "URN5678",
+                            "party": [
```
Comment (KianKwa) [inline src/test/resources/mocks/opaPublicList.json:143]: This group of party just like the previous case. Does not need to repeat the party here. You can probably remove some of the party here

File: src/test/resources/mocks/opaPressList.json (hunk 3 above)
Comment (MindGames2015) [reply src/test/resources/mocks/opaPressList.json:69]: Removed 2 Defendants. Thanks Kian

File: src/test/resources/mocks/opaPublicList.json (hunk 5 above)
Comment (MindGames2015) [reply src/test/resources/mocks/opaPublicList.json:143]: Removed repeating part for "caseUrn": "URN5678",

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPressListHelper.java (hunk 2 above)
Comment (MindGames2015) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPressListHelper.java:111]: Reverted method formatReportingRestriction in both OPA press/public to use Generic method. Thanks Kian

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPublicListHelper.java (hunk 1 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPublicListHelper.java:85]: Hey Paul I can still see the formatReportingRestriction method here

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPressListHelper.java (hunk 2 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/listmanipulation/OpaPressListHelper.java:111]: formatReportingRestriction is still being used

**Review by KianKwa — APPROVED** [review]

**Pull request merged**